test/threads.jl

```diff
@@ -141,3 +141,41 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic fences
+type CommBuf
+    barrier::Atomic{Int}
+    data::Vector{Float64}
+    correct::Bool
+    CommBuf() = new(Atomic{Int}(0), Vector{Float64}(100), false)
+end
+function test_fence_write(commbuf::CommBuf)
+    sleep(0.1)
```

Comment:
I think the safest is to just ensure a small delay. I'm thinking of busy-waiting via `time()`.